prime/src/data/actions/Actions.ts

```diff
@@ -40,7 +40,24 @@ export interface AccountState {
   readonly claimedFeeUniswapKeys: readonly string[];
 }
 
-type Operator = (state: AccountState) => AccountState | null;
+export type HypotheticalAccountStates = {
+  accountStates: AccountState[];
+  errorMsg?: string;
+};
+
+export type SuccessResult = {
+  success: true;
+  accountState: AccountState;
+};
+
+export type ErrorResult = {
+  success: false;
+  error: Error;
+};
+
+export type OperationResult = SuccessResult | ErrorResult;
```

Comment:
Fun new types allow us to define success and have it dictate what else is present.